### PR TITLE
Added documentation/general_docs/building

### PR DESCRIPTION
**Pull request by BobbyRBruce**

### CODE REVIEW

File: _pages/documentation/general_docs/building/index.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+# Building gem5
+
+## Dependencies
+
+* **git** : gem5 uses git for version control
+* **gcc 4.8+**: gcc is used to compiled gem5. Version 4.8+ must be used. We do
+presently support > Version 7. (Note: [Support for gcc 4 may be dropped](
+https://gem5.atlassian.net/browse/GEM5-218), and [version >7 may be
+supposed](https://gem5.atlassian.net/browse/GEM5-194) in future releases of
```
Comment (powerjg) [inline _pages/documentation/general_docs/building/index.md]: supposed -> supported?

File: _pages/documentation/general_docs/building/index.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+gem5.
+* SCons : gem5 uses SCons as its build environment.
+* Python 2.7+ : gem5 replies on Python development libraries (due to the
+[retirement of Python 2](
+http://pyfound.blogspot.com/2019/12/python-2-sunset.html) we are [likely to
+migrate to Python 3 in future releases of gem5](
+https://gem5.atlassian.net/browse/GEM5-275).
+* protobuf 2.1+ : The protobuf library is used for trace generation and
```
Comment (powerjg) [inline _pages/documentation/general_docs/building/index.md]: IIRC, this is optional, but I could be wrong.

File: _pages/documentation/general_docs/building/index.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+gem5's build system is based on SCons, an open source build system implemented
+in Python. You can find more information about scons at <http://www.scons.org>.
+The main scons file is called SConstruct and is found in the root of the source
+tree. Additional scons files are named SConscript and are found throughout the
+tree, usually near the files they're associated with.
+
+Within the root of the gem5 directory, gem5 can be built with SCons using:
+
+```
+scons build/{ISA}/gem5.{variant}
```
Comment (powerjg) [inline _pages/documentation/general_docs/building/index.md]: Should probably mention `-j <cpus>` here, tool. Additionally, giving an estimate for how long this takes would be good.

File: _pages/documentation/general_docs/building/index.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+```
+
+where `{ISA}` is the target Instruction Set Architecture, and `{variant}`
```
Comment (powerjg) [inline _pages/documentation/general_docs/building/index.md]: `is the target` -> `is the target (guest)`

File: _pages/documentation/general_docs/building/index.md (hunk 1 above)
Comment (BobbyRBruce) [reply _pages/documentation/general_docs/building/index.md]: Done

File: _pages/documentation/general_docs/building/index.md (hunk 3 above)
Comment (BobbyRBruce) [reply _pages/documentation/general_docs/building/index.md]: Done

File: _pages/documentation/general_docs/building/index.md (hunk 4 above)
Comment (BobbyRBruce) [reply _pages/documentation/general_docs/building/index.md]: Done

File: _pages/documentation/general_docs/building/index.md (hunk 2 above)
Comment (BobbyRBruce) [reply _pages/documentation/general_docs/building/index.md]: Done

**Comment by BobbyRBruce** [comment]
arg, i have no idea what I've done here to cause these conflicts

**Comment by BobbyRBruce** [comment]
To avoid these conflicts (which I couldn't figure out). I recreated these changes on a new branch and merged successfully: https://github.com/gem5/new-website/pull/61 .

**Pull request closed without merging**